instrumentation/jetty-httpclient/jetty-httpclient-9.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_0/JettyHttpClient9InstrumentationModule.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_0;
+
+import static io.opentelemetry.javaagent.extension.matcher.ClassLoaderMatcher.hasClassesNamed;
+import static io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_0.JettyHttpClient9Tracer.tracer;
+import static java.util.Collections.singletonList;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.util.List;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.eclipse.jetty.client.api.Request;
+
+@AutoService(InstrumentationModule.class)
+public class JettyHttpClient9InstrumentationModule extends InstrumentationModule {
+
+  public JettyHttpClient9InstrumentationModule() {
+    super("jetty-httpclient", "jetty-httpclient-9");
+  }
+
+  @Override
+  public List<TypeInstrumentation> typeInstrumentations() {
+    return singletonList(new JettyHttpClient9Instrumentation());
+  }
+
+  public static class JettyHttpClient9Instrumentation implements TypeInstrumentation {
+
+    @Override
+    public ElementMatcher<ClassLoader> classLoaderOptimization() {
+      return hasClassesNamed("org.eclipse.jetty.client.api.Request$BeginListener");
+    }
+
+    @Override
+    public ElementMatcher<TypeDescription> typeMatcher() {
+      return hasSuperType(named("org.eclipse.jetty.client.api.Request").and(isInterface()));
+    }
+
+    @Override
+    public void transform(TypeTransformer transformer) {
+      transformer.applyAdviceToMethod(
+          isMethod().and(named("send")),
+          JettyHttpClient9InstrumentationModule.class.getName() + "$JettyHttpClient9Advice");
+    }
+  }
+
+  public static class JettyHttpClient9Advice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void addTracingEnter(
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope,
+        @Advice.This Request jettyRequest) {
+
+      Context parentContext = currentContext();
+      if (!tracer().shouldStartSpan(parentContext)) {
+        return;
+      }
+
+      JettyHttpClient9TracingInterceptor interceptor =
```

Comment:
Usually we call a static method on tracer that starts the span and returns `Context` and then call `makeCurrent` on it to set up scope. Looking at `attachToRequest` it seems that `getCtx` can return null, if that is the case you need to perform a null check before calling `makeCurrent`. Also we use verbose names so `getContext` instead of `getCtx` would be a better.

Reply:
Thanks @laurit I'll definitely remove the Null possibility and name appropriately

Reply:
Context verbose naming is in place

Reply:
should be done now